.github/workflows/main.yml

```diff
@@ -1,22 +1,27 @@
-name: Merge queue and try
+name: Main
 
 on:
   push:
-    branches: ["auto", "try"]
-
-  # Used by GH merge queue for landing PRs
+    # Run the entire pipeline for 'master' even though the merge queue already runs checks
+    # for every change. This just offers an extra layer of testing and covers the case of
+    # random force pushes.
+    branches: ["master", "try"]
+  pull_request:
+    types: ['opened', 'synchronize']
+    branches: ["**"]
   merge_group:
     types: [checks_requested]
-
-  # Allows you to run this workflow manually from the Actions tab
   workflow_dispatch:
 
 jobs:
   decision:
-    name: Decision job
+    name: Decision
     runs-on: ubuntu-20.04
     steps:
-      - name: Previous successful build check
+      # If an identical build exists that suceeded, skip this workflow run. We don't do
+      # this check for pull_request and merge_group events, out of caution.
+      - name: Skip previous identical builds
+        if: ${{ !github.event_name != 'pull_request' && github.event_name != 'merge_group' }}
```

Comment:
`!github.event_name != 'pull_request'`

should this be just `github.event_name != 'pull_request'` without the negation at the front?

Reply:
It should be without the leading `!`. I've pushed a fix for that. It's interesting that GitHub didn't complain.